pkg/mqtrigger/messageQueue/kafka/consumer.go

```diff
@@ -200,7 +200,7 @@ func (ch *MqtConsumerGroupHandler) kafkaMsgHandler(msg *sarama.ConsumerMessage)
 		errorString := fmt.Sprintf("request exceed retries: %v", ch.trigger.Spec.MaxRetries)
 		errorHeaders := generateErrorHeaders(errorString)
 		errorHandler(ch.logger, ch.trigger, ch.producer, ch.fnUrl,
-			fmt.Errorf(errorString), errorHeaders)
+			fmt.Errorf("%s", errorString), errorHeaders)
```

Comment:
Should be errors.New if we are doing formatting. Same for rest. Lets avoid using formatting unnecessarily.